pkg/auth/requests/authenticate.go

```diff
@@ -112,17 +118,28 @@ func (a *tokenAuthenticator) Authenticate(req *http.Request) (*AuthenticatorResp
 		return nil, errors.Wrapf(ErrMustAuthenticate, "clusterID does not match")
 	}
 
+	// If the auth provider is specified make sure it exists and enabled.
+	if token.AuthProvider != "" {
+		disabled, err := providers.IsDisabledProvider(token.AuthProvider)
+		if err != nil {
+			return nil, errors.Wrapf(ErrMustAuthenticate, "error checking if provider %s is disabled: %v", token.AuthProvider, err)
+		}
+		if disabled {
+			return nil, errors.Wrapf(ErrMustAuthenticate, "provider %s is disabled", token.AuthProvider)
+		}
+	}
+
 	attribs, err := a.userAttributeLister.Get("", token.UserID)
 	if err != nil && !apierrors.IsNotFound(err) {
-		return nil, err
+		return nil, errors.Wrapf(ErrMustAuthenticate, "failed to retrieve userattribute %s: %v", token.UserID, err)
```

Comment:
I can't seem to find a reason why `err` wasn't wrapped with `ErrMustAuthenticate` before.